pkg/operator/upgradebackupcontroller/upgradebackupcontroller_test.go

```diff
@@ -0,0 +1,146 @@
+package upgradebackupcontroller
+
+import (
+	"context"
+	"fmt"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	u "github.com/openshift/cluster-etcd-operator/pkg/testutils"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	"testing"
+	"time"
+)
+
+func Test_ensureRecentBackup(t *testing.T) {
+	scenarios := []struct {
+		name                    string
+		pods                    []*corev1.Pod
+		clusterversionCondition configv1.ClusterOperatorStatusCondition
+		wantBackupStatus        operatorv1.ConditionStatus
+		wantNilBackupCondition  bool
+		wantErr                 bool
+	}{
+		{
+			name: "failed backup pod",
+			pods: []*corev1.Pod{
+				u.FakePod("cluster-backup", u.WithPodStatus(corev1.PodFailed), u.WithCreationTimestamp(nowMinus(40*time.Second))),
+			},
+			clusterversionCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    "Failing",
+				Status:  configv1.ConditionTrue,
+				Message: fmt.Sprintf("Need RecentBackup"),
+			},
+			wantBackupStatus: operatorv1.ConditionFalse,
+		},
+		{
+			name: "backup pod pending status",
+			pods: []*corev1.Pod{
+				u.FakePod("cluster-backup", u.WithPodStatus(corev1.PodPending)),
+			},
+			clusterversionCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    "Failing",
+				Status:  configv1.ConditionTrue,
+				Message: fmt.Sprintf("Need RecentBackup"),
+			},
+			wantBackupStatus: operatorv1.ConditionUnknown,
+		},
+		{
+			name: "RecentBackup not required invalid type",
+			clusterversionCondition: configv1.ClusterOperatorStatusCondition{
+				Type:    "NotFailing",
+				Status:  configv1.ConditionTrue,
+				Message: fmt.Sprintf("Need RecentBackup"),
```

Comment:
if this is invalid type should this not fail?

Reply:
well the idea here is to not process the operator unless a valid combination of Type and Message is observed. In this case we have the valid message but not the valid type. So we return nil as it is a no-opp (skip)